CMakeLists.txt

```diff
@@ -19,15 +19,14 @@ if (BUILD_TARGET MATCHES "gpu")
   option(ENABLE_CUDA "Allow user to activate/deactivate CUDA support" ON)
   option(ENABLE_MANAGED "Allow user to activate/deactivate automatic memory managment from NVHPC" OFF)
   option(ENABLE_NCCL "Allow user to activate/deactivate Collective Comunication NCCL" OFF)
-
   if (ENABLE_CUDA)
-    message(STATUS "Before enable CUDA")
     enable_language(CUDA)
-    message(STATUS "After enable CUDA")
   endif()
-
 endif(BUILD_TARGET MATCHES "gpu")
 
+include(CMakeDependentOption)
+cmake_dependent_option(ENABLE_OMP "Use OpenMP with pure MPI" ON "NOT ENABLE_CUDA" OFF)
```

Comment:
Default value for ENABLE_OMP should be off ?

Reply:
I put this on and removed only in case of GPU compilation. Now all compilers support OpenMP therefore should be fine to have it on. The only trick is that the num threads needs to set. For our testing it is done. We need a note that if someone is using it with another application they need to do it.

Reply:
I think the default should be OpenMP off : the library was designed without OpenMP initially. Setting the default OpenMP to off will not lead to perturbations for external codes currently using 2decomp/main and switching to 2decomp/dev.

Reply:
I agree, I think we should try and keep the behaviour the same as before (pure MPI) and allow users to enable threading (MPI+X) if desired

Reply:
Ok I'll put it to OFF by default.

Reply:
I just pushed the OFF by default. I checked and it can be activated also using `ccmake` without the need to have it in the configure line.